kaska/s2_observations.py

```diff
@@ -219,133 +263,181 @@ def read_time_series(self, time_grid):
         return s2_obs
 
     def read_granule(self, timestep):
-        """Reads data granule for a given `timestep`. Returns all relevant
-        bits and bobs (surface reflectrance, angles, cloud mask, uncertainty).
+        """Reads the data for the given timestep. Returns all relevant
+        quantities (surface reflectrance, angles, cloud mask, uncertainty).
         The mask is true for OK pixels. If there are no suitable pixels, the
         returned tuple is a collection of `None`
 
-
         Parameters
         ----------
-        timestep : datetime
-            The datetime object
-
+        timestep : A datetime object.
         Returns
         -------
-        tuple
-            rho_surface, mask, sza, vza, raa, rho_unc
-        """
+        tuple containing the rho_surface, mask, sza, vza, raa, rho_unc
 
+        NOTE: Currently reads in sequentially. It's better to gather
+        all the filenames and read them in parallel using parmap.py
+        """
         assert timestep in self.date_data, f"{str(timestep):s} not available!"
-        # NOTE: Currently reads in sequentially. It's better to gather
-        # all the filenames and read them in parallel using parmap.py"""
-        current_folder = self.date_data[timestep]
-        fname_prefix = [
-            f.name.split("B02")[0] for f in current_folder.glob("*B02_sur.tif")
-        ][0]
-        # Find cloud mask
-        cloud_mask = current_folder.parent / f"cloud.tif"
+
+        # This is the parent folder for this granule
+        current_folder = Path()
+        for part in self.date_data[timestep][0].parts:
+            current_folder = current_folder / part
+            if ".SAFE" in part:
+                break
```

Comment:
If the xml files are correct, there's always a `.SAFE` directory in the path, because of the way the file list has been created. But I put an assert there anyways - who ever messes up with xml metadata files????